app/controllers/maintenances_controller.rb

```diff
@@ -1,2 +1,69 @@
 class MaintenancesController < ApplicationController
+  before_action :set_maintenance, only: %i[show edit update destroy]
+  before_action :set_vehicle, only: %i[index create edit update]
+
+  def index
+    populate_index
+
+    @maintenance = Maintenance.new
+    @btn_txt = "Ajouter"
+  end
+
+  def create
+    @maintenance = Maintenance.new(maintenance_params)
+    @maintenance.vehicle_id = params[:vehicle_id]
+
+    if @maintenance.save
+      redirect_to vehicle_maintenance_operations_path(params[:vehicle_id], @maintenance)
+    else
+      @btn_txt = "Ajouter"
+      populate_index
+      render "maintenances/index"
+    end
+  end
+
+  def edit
+    @btn_txt = 'Editer'
+    populate_index
+    render "maintenances/index"
+  end
+
+  def update
+    @maintenance.vehicle_id = params[:vehicle_id]
+    if @maintenance.update(maintenance_params)
+      redirect_to vehicle_maintenances_path(params[:vehicle_id])
+    else
+      @btn_txt = "Editer"
+      populate_index
+      render "maintenances/index"
+    end
+  end
+
+  def destroy
+    @maintenance.destroy
+
+    redirect_to vehicle_maintenances_path(params[:vehicle_id])
```

Comment:
Using the added before action to use the instance variable
```suggestion
    redirect_to vehicle_maintenances_path(@vehicle)
```